File: CS-Examples/21_Security/DetectProtection.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace DetectProtection

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Specify the path of the input file
            Dim input As String = "..\..\..\..\..\..\Data\ProtectedWorkbook.xlsx"

            ' Check if the workbook at the specified path is password protected
            Dim value As Boolean = Workbook.IsPasswordProtected(input)

            ' If the workbook is password protected, set the text of textBox1 to "Yes"
            If value Then
                textBox1.Text = "Yes"
            Else
                ' If the workbook is not password protected, set the text of textBox1 to "No"
                textBox1.Text = "No"
            End If

        End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/21_Security/DetectProtection.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace DetectProtection
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            // Specify the input file
            string input = @"..\..\..\..\..\..\Data\ProtectedWorkbook.xlsx";

            //Detect if the Excel workbook is password protected
            bool value = Workbook.IsPasswordProtected(input);

            if (value)
            {
                textBox1.Text = "Yes";
            }
            else
            {
                textBox1.Text = "No";
            }

		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
